### ISSUE TITLE
File download does not work in IE 5.5 SP1 on Win95

### ISSUE THREAD

**Issue by sekkingsoft** [ttalvitie/browservice#69]
Browservice seems to be working pretty much flawlessly on Windows 95 OSR2.5 with Internet Explorer 5.5 SP1, with one major exception; File download does not work. The download button will show, the file will download to the browservice server, but IE will only download a HTML file with a scrambled file name, seemingly the same HTML page as is currently shown in the browser (the browservice client).

IE version 5.50.4522.1800
Windows 95 OSR2.5
Browservice 0.9.4.0 armhf running on an Raspberry Pi 3 Model B

File download works perfectly fine in Netscape 7.2 and Firefox 1.0

![browservice](https://user-images.githubusercontent.com/59697679/157114610-b2b07e77-c56d-4c11-a35a-80bb9c9eb65e.png)

![internetexplorer_ver](https://user-images.githubusercontent.com/59697679/157114349-b89ba051-50ef-4826-89b0-9a1c24f1b66f.png)

**Comment by ttalvitie** [comment]
Did you actually check that the downloaded .html file is a HTML page, instead of the file you were trying to download under a wrong name and extension? Old IEs on old Windows versions are weird about downloads, even when not using Browservice; they simply force some file downloads to have the .html extension but if you change it the file itself will be correct. I don't see why the browser would download the client web page instead; if changing the extension doesn't work, could you show an example (complete with screenshots) of downloading a publicly available file and getting wrong file contents?

**Comment by sekkingsoft** [comment]
Yes, it is indeed a HTML page. I will find an example later today. I should also mention that IE loses it's connection to browservice after downloading the file.

Upload works great tho. This github issue is created with this setup :)

**Comment by nguyenthienminh** [comment]
I had the exact setup as yours and well, the file that I wanted to download downloaded successfully.

**Comment by nguyenthienminh** [comment]
Seems like the exact same thing on my side. There's nothing wrong there.

**Comment by sekkingsoft** [comment]
I have tried multiple files, this was just an example as it was easy to find.

**Comment by ttalvitie** [comment]
I was also able to reproduce this on MSIE 5.50.4522.1800 (English) on Windows 95. However, the issue does not appear on any of the following versions I tested:
- MSIE 5.0: 5.00.3314.2101
- MSIE 5.5 (without SP): 5.50.4134.0600
- MSIE 5.5 SP2: 5.50.4807.2300
It seems that for some reason in this particular version, MSIE 5.5 SP1, the browser does not handle downloads from iframes correctly, instead downloading the parent page HTML. Because it only seems to affect the SP1 version and not the original or the SP2 version, I think it's really not worth it to try to fix it; the fix would most likely require major changes to the download logic and risk breaking compatibility with other client browsers. 

I suggest that you simply upgrade to SP2 (or downgrade to pre-SP1, if possible).
